### PR TITLE
Update version 0.3.0

### PR DESCRIPTION
**Pull request by cbellot000**
In this Pull Request all the update from client 0.2.1 to client 0.3.0 are pushed. 
Changes are made:
- issues fixing
- server updates
- new developments

### CODE REVIEW

**Comment by akaszynski** [comment]
@cbellot000, you're going to have to update both the docker and windows package (currently stored as a universal artifact) to get this to run correctly.  We've implemented backwards compatibility for older images here in the PyMAPDL repo:
https://github.com/pyansys/PyMAPDL/blob/master/.ci/azure-pipelines.yml

Effectively, we're running two jobs, (one for the older version and one for the newer).

I'll be able to start reviewing this PR once the unit testing is passing for both the old version and the new version of DPF.

File: ansys/dpf/core/collection.py [hunk 1]
```diff
@@ -52,65 +66,103 @@ def set_labels(self, labels):
...
         return self._info['labels']
 
     labels = property(_get_labels, set_labels, "labels")
+    
+    def has_label(self, label):
+        """Check if a collection has a specified label
+
+        Parameters
+        ----------
+        label: string
```
Comment (akaszynski) [inline ansys/dpf/core/collection.py:126]:
```suggestion
        label: str
```

File: ansys/dpf/core/collection.py [hunk 2]
```diff
@@ -52,65 +66,103 @@ def set_labels(self, labels):
...
+            Labels that must be searched, for example ``'time'``.
+            
+        Returns
+        -------
+        bool: True if the specified value have been found in the collection.
```
Comment (akaszynski) [inline ansys/dpf/core/collection.py:131]:
```suggestion
        bool
            ``True`` if the specified value has been found in the collection. 
```

File: ansys/dpf/core/collection.py [hunk 3]
```diff
@@ -52,65 +66,103 @@ def set_labels(self, labels):
...
+
+        Examples
+        --------
+        >>> from ansys.dpf import core as dpf
+        >>> coll = dpf.Collection(dpf.types.field)
+        >>> coll.add_label('time')
+        >>> coll.has_label('time')
+        True
+        >>> coll.has_label('complex')
+        False
```
Comment (akaszynski) [inline ansys/dpf/core/collection.py:141]:
```suggestion

        >>> coll.has_label('complex')
        False
```
Need to have an empty line for this to be rendered correctly in sphinx.

File: ansys/dpf/core/collection.py [hunk 4]
```diff
@@ -52,65 +66,103 @@ def set_labels(self, labels):
...
+        
+        """
+        result = False
+        labels = self.labels
+        i = 0
+        while i < len(labels):
+            if labels[i] == label:
+                result = True
+            i += 1
+        return result
```
Comment (akaszynski) [inline ansys/dpf/core/collection.py:151]:
```suggestion
        return label in self.labels
```
https://stackoverflow.com/questions/7571635/fastest-way-to-check-if-a-value-exists-in-a-list

**Review by akaszynski — APPROVED** [review]
Remove `ansys/dpf/__init__.py` and we're good to go.  I'll take another look once merged for a cleanup as this is too big to review effectively.  Unit tests pass and docbuild looks good.

**Comment by cbellot000** [comment]
Hi @akaszynski, is that wanted that we can't create a merge commit but we can only create a squash merge? Since this PR is the result of many others in the other repo, I kept the history when cherry picking in the other repo

**Pull request merged**